### PR TITLE
Authentication + Registration

### PR DESCRIPTION
**Pull request by tomtomecek**
* First part is about test setup.
* Second part is about building authentication and registration for users. Couple feature specs to capture requirements logic and couple controller specs to capture restrictions and auth requirements.

+ Admin can add/edit/remove/hide/show questions and UI is following this and hides stuff from non-admin.

### CODE REVIEW

File: spec/support/expectation_macros.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module ExpectationMacros
+  def expect_to_see(text, options = {})
+    expect(page).to have_content(text, options)
+  end
+
+  def expect_to_not_see(text, options = {})
+    sleep 0.1
+    expect(page).to have_no_content(text, options)
+  end
+  alias_method :expect_not_to_see, :expect_to_see
```
Comment (knwang) [inline spec/support/expectation_macros.rb]: mind blown

**Comment by knwang** [comment]
looks good to me :) where's the deployed app?

**Comment by tomtomecek** [comment]
http://tl-interview-questions.herokuapp.com - but the changes has not been pushed to production yet. I have 2 PRs incoming shortly after this is merged.

**Comment by knwang** [comment]
ok sounds good.

**Pull request merged**